File: src/tractor/Call.java
package tractor;

public class Call {
	Card card;
	int count;
	public Call (Card card, int count) {
		this.card = card;
		this.count = count;
	}
	public Call (String cardString, int count) {
		this.card = new Card(cardString);
		this.count = count;
	}
	public int getCallLevel() {
		int callLevel = card.getRawSuit() + (count - 1) * 6;
		if (card.toString().equals("J+")) {
			callLevel++;
		}
		if (callLevel == 4 || callLevel == 5) {
			callLevel = -1;
		}
		return callLevel;
	}
	public int getSuit() {
		return card.getRawSuit();
	}
}

File: src/python/Call.py
from Card import Card

class Call:

  def __init__(self, card, count):
    if type(card) == Card:
      self.card = card
    else:
      self.card = Card(card)
    self.count = count

  def getCallLevel(self):
    callLevel = self.card.getRawSuit() + (self.count - 1) * 6
    if self.card.toString() == "J+":
      callLevel += 1
    if callLevel == 4 or callLevel == 5:
      callLevel = -1
    return callLevel
  
  def getSuit(self):
    return self.card.getRawSuit()
    
  def isValid(self):
    return self.card.getPowerIndex() >= 26
  
  def toString(self):
    return str(self.count) + " of " + self.card.toString()
  
  def serialize(call):
    return call.card.toString() + "," + str(call.count)
  
  def serializeSelf(self):
    return Call.serialize(self)
  
  def deserialize(callString):
    card = Card(callString[:callString.index(",")])
    count = int(callString[callString.index(",") + 1:])
    return Call(card, count)